### PR TITLE
ContextRDD Tree Aggregate and Aggregate

### PR DESCRIPTION
**Pull request by danking**
cc: @cseed

### CODE REVIEW

**Comment by danking** [comment]
it's not tested

**Comment by danking** [comment]
I'll write a test

**Comment by danking** [comment]
I had to add `TrivialContext` which is just a dummy context for when I'm not really using regions. I guess I could also add tests that use the context in some meaningful way. The tension is that the context-using-functionality should already be tested via MatrixTable and Table.

File: src/test/scala/is/hail/sparkextras/ContextRDDSuite.scala [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  @Test
+  def aggregateDistinctTandUTypes() {
+    for {
+      (name, data) <- Seq(
+        "indivisble number of partitions" ->
+          ContextRDD.parallelize[TrivialContext](
+            sc,
+            (0 until 256).map(Array.fill(_)(0)),
+            5),
+        "divisble number of partitions" ->
```
Comment (jigold) [inline src/test/scala/is/hail/sparkextras/ContextRDDSuite.scala]: These two tests are identical. Can you also add single partition test?

File: src/test/scala/is/hail/sparkextras/ContextRDDSuite.scala [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    }
+  }
+
+  @Test
+  def treeAggregateDistinctTandUTypes() {
+    for {
+      (name, data) <- Seq(
+        "indivisble number of partitions" ->
+          ContextRDD.parallelize[TrivialContext](
+            sc,
```
Comment (jigold) [inline src/test/scala/is/hail/sparkextras/ContextRDDSuite.scala]: Same data being tested. Also add partition size == 1.

File: src/test/scala/is/hail/sparkextras/ContextRDDSuite.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+          ContextRDD.parallelize[TrivialContext](sc, 0 until 256, 16),
+          2
+        ),
+        ("lots of records, divisble number of partitions, depth 3",
+          ContextRDD.parallelize[TrivialContext](sc, 0 until 256, 16),
+          3
+        ),
+        ("lots of records, indivisble number of partitions, depth 3",
+          ContextRDD.parallelize[TrivialContext](sc, 0 until 256, 15),
+          2
```
Comment (jigold) [inline src/test/scala/is/hail/sparkextras/ContextRDDSuite.scala]: depth should be 3 here

**Pull request merged**